src/lib/yang/schema.lua

```diff
@@ -540,6 +540,40 @@ declare_initializer(init_typedef, 'typedef')
 declare_initializer(init_uses, 'uses')
 declare_initializer(init_value, 'value')
 
+
+-- 7.8.2.  The list's key Statement
+
+-- The "key" statement, which MUST be present if the list represents
+-- configuration, and MAY be present otherwise, takes as an argument a
+-- string that specifies a space-separated list of leaf identifiers of
+-- this list.  A leaf identifier MUST NOT appear more than once in the
+-- key.  Each such leaf identifier MUST refer to a child leaf of the
+-- list.  The leafs can be defined directly in substatements to the
+-- list, or in groupings used in the list.
+local function sanitize_lists (node)
+   if node.kind and node.kind == "list" and node.key then
+      assert(node.body)
```

Comment:
Don't need to check kind exists in lua afaik, simply doing `if node.kind == "list"` is sufficient as `node.kind` will be `nil` if it doesn't exist.